### ISSUE TITLE
Build "Add Your Club" workflow

### ISSUE THREAD

**Issue by hannahkane** [mozilla/teach.webmaker.org#71]
### RACI
* Phase: Build
* Owner: @hannahkane 
* Decision Maker: @hannahkane 
* Design Lead: NA
* Dev Lead: ?
* Quality: @davidascher 

See Design ticket: #70 

### Requirements
* Responsive
* Works in all supported browser versions
* Store submitted Club information and display on map
* Should require log-in or sign-up, but should be as seamless as possible
* Should trigger transactional email (see email number 2 here: https://teach.etherpad.mozilla.org/clubtransactionalemails)

Tasks:
 - [ ] Design review
 - [ ] Content review

**Comment by hannahkane** [comment]
Question for @adamlofting — looking for recommendations about how to handle email capture here. Ideally, the "add your club" workflow is not dependent on the user having a Teach account (see #46). I'm thinking that means we'll need to add an email opt-in to the "Add your club" form so that we can email users who add their Club, but who haven't signed up for a Teach account. 

Or perhaps the simplest solution is to require a Teach account to complete the "Add Your Club" workflow. That feels overly burdensome, though. :(

**Comment by hannahkane** [comment]
Realizing the sign-up flow might not be any more burdensome than the Add Your Club flow, so maybe this point is moot. Maybe adding your Club *is* signing up—we just need to make that transparent to the user.

**Comment by adamlofting** [comment]
#### TL;DR: I vote for getting people to create accounts.

--
My view (though not based on any experimental metrics :wink:) is that the most long term value comes from getting people into the overall social-engaging-interactive bit of joining this community as quickly as possible. In the long run, that's creating accounts and sending relevant transactional emails about the things they are doing and could do.

--

An "account creation" flow can be as simple as an "email sign-up" flow. And either will be small in relation to extra steps of an "Add your club" flow. 

Also, by the time someone is going to run a Club, small UX performance tweaks on the sign-up form are not going to have a significant impact, because they are strongly predisposed to completing the task.

--

After account creation, users can flag things to us about their interests / skills / location etc that help us send them relevant asks and content. This becomes more like a user-managed CRM profile than a basic email list. And allows us to build connections between between with similar or complimentary skills and objectives. Building this into the user profiles early could be really useful.

--

We need to note two types of email communications:
* Transactional product emails, limited to the scope of the product being used do not need additional opt-in (though there must be a way to opt-out)
  * E.g. A notification to a webmaker user to say their make has been remixed 10 times
* General Mozilla communication emails do need the opt-in
  * E.g. Emailing a webmaker user to ask them to sign a Net Neutrality petition

I think your priority is transactional emails related to the Mozilla Learning experience. And an additional email opt-in to broader Mozilla comms is a nice extra to support broader Mozilla engagement efforts. 

To add some actual numbers to this conversation. Around 50% of people who create a webmaker account opt-in to receive general email updates from Mozilla. Dashboard is here: https://metrics.webmaker.org/dashboards

--

One very last thought, as this comment is waaaaaay too long... On day 1, this thing doesn't have to work and have a real database and do everything nicely. It can be really basic and manual behind the scenes while we test and prove how users actually use the thing. If the anticipated initial volume is in the 10s, rather than the 100s we can even have people email us and manually add them to a map while we learn about their needs. A google sheet of the first 50 or so manually entered users can be migrated to a database pretty quickly.

**Comment by hannahkane** [comment]
Thanks, @adamlofting! 

Just updated the ticket description to specify that Add Your Club flow should require authentication. The sign-up flow described in #46 is really quite simple, as you suggest, so I agree that it won't be too much of a burden (though I do want to include the opt-in for Mozilla comms, as my understanding is that's very important to us).

I'd originally had a full user profile included in v1, but made the decision to postpone it, since it's not fundamental to our initial goals. (See #39.) We're on the same page about the (eventual) importance of that.

Re: the issue of a real dbase—that's true, and I'll leave it up to the developers to decide what's possible within this sprint. Seems to me we're well-resourced this sprint, and, aside from this flow and the regular sign-up flow, most of the other tasks are building static pages, so I hope we can do it. :)

**Comment by vazquez** [comment]
#### Assets

Globe with Pin: https://www.dropbox.com/s/xshn444bw1llg83/globe-with-pin.svg?dl=0

Envelope: https://www.dropbox.com/s/jh6i06h8670n545/envelope.svg?dl=0

**Comment by toolness** [comment]
This has been done in #354!

Note that login is still done via persona,  but we're associating the users with their actual Webmaker accounts, so migration to the new login mechanism shouldn't be too hard.

I'm not sure if this issue should be closed or not though so I will leave it up to @hannahkane to do that.

**Comment by hannahkane** [comment]
yay!

Will close this and open a different ticket about hooking up the new login/signup experience to the Clubs workflow.
